Core/include/Acts/TrackFitting/detail/GsfUtils.hpp

```diff
@@ -136,13 +136,54 @@ class ScopedGsfInfoPrinterAndChecker {
   }
 };
 
+ActsScalar calculateDeterminant(
+    TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
+                     true>::Measurement fullCalibrated,
+    TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
+                     true>::MeasurementCovariance fullCalibratedCovariance,
+    TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
+                     true>::Covariance predictedCovariance,
+    TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax, true>::Projector
+        projector,
+    unsigned int calibratedSize);
+
 /// Reweight the components according to `R. Frühwirth, "Track fitting
 /// with non-Gaussian noise"`. See also the implementation in Athena at
 /// PosteriorWeightsCalculator.cxx
 /// @note The weights are not renormalized!
-void computePosteriorWeights(const MultiTrajectory &mt,
-                             const std::vector<std::size_t> &tips,
-                             std::map<std::size_t, double> &weights);
+template <typename D>
+void computePosteriorWeights(
+    const MultiTrajectory<D> &mt,
+    const std::vector<MultiTrajectoryTraits::IndexType> &tips,
+    std::map<MultiTrajectoryTraits::IndexType, double> &weights) {
+  // @TODO: Put back into compilation unit
```

Comment:
todo?

Reply:
Probably not necessary. Removed the TODO.